src/utilities/ntrace/TracePluginImpl.cpp

```diff
@@ -1907,6 +1910,24 @@ bool TracePluginImpl::checkServiceFilter(ITraceServiceConnection* service, bool
 }
 
 
+void TracePluginImpl::checkErrno(const char* operation)
+{
+	if (errno == 0)
+		return;
+
+	const char* strErr;
+#ifdef WIN_NT
+	strErr = strerror(errno);
+#else
+	char buff[256];
+	strerror_r(errno, buff, sizeof(buff));
+	strErr = buff;
+#endif
+	fatal_exception::raiseFmt("TracePluginImpl: operation \"%s\" failed on file \"%s\". Error is : %s",
+		operation, config.log_filename.c_str(), strErr);
+}
+
+
```

Comment:
Why fatal_exception ? It kill whole Firebird process, iirc. Too much, i think.
Why not OS native error code and corresponding exception ?
Explain, please: how write failure is related with "creation of trace log file fails" ?

Reply:
The exception is caught in TraceFactoryImpl::trace_create(...), like in FB 2.5. Trace log file is created during the first write attempt.